tests/test_frozen_layers.py

```diff
@@ -0,0 +1,187 @@
+from torch import nn
+
+from nncf import NNCFConfig
+from tests.helpers import TwoConvTestModel, create_compressed_model_and_algo_for_test
+from tests.quantization.test_quantization_helpers import get_quantization_config_without_range_init
+from tests.pruning.helpers import get_basic_pruning_config
+from tests.sparsity.rb.test_algo import get_basic_sparsity_config
+from nncf.layers import NNCF_MODULES_MAP, NNCF_MODULES_DICT
+from typing import NamedTuple
+from nncf.utils import get_module_by_node_name
+
+
+def freeze_module(model, scope=None):
+    if scope is not None:
+        for nncf_module in NNCF_MODULES_MAP.keys():
+            if nncf_module in scope:
+                scope = scope.replace(nncf_module, NNCF_MODULES_MAP[nncf_module])
+                break
+        module = get_module_by_node_name(model, scope)
+        module.weight.requires_grad = False
+        return
+    for module in model.modules():
+        if module.__class__ in NNCF_MODULES_DICT.values():
+            module.weight.requires_grad = False
+            break
+
+
+def update_config(config, update_dict):
+    config.update(update_dict)
+    return config
+
+
+class FrozenLayersTestStruct(NamedTuple):
+    config: 'NNCFConfig' = get_quantization_config_without_range_init
+    model_creator: Callable[[], nn.Module] = TwoConvTestModel
+    is_error: bool = True
+
+
+TEST_PARAMS = [
+    FrozenLayersTestStruct(config=update_config(get_quantization_config_without_range_init(), {
+        "compression": {
+            "algorithm": "quantization"
+        },
+        "ignored_scopes": ['TwoConvTestModel/Sequential[features]/Sequential[0]/Conv2d[0]']
+    })
+    , model_creator=TwoConvTestModel, is_error=False),
+    FrozenLayersTestStruct(config=update_config(get_quantization_config_without_range_init(), {
+        "target_device": "VPU",
+        "compression": {
+            "algorithm": "quantization",
+            "initializer": {
+                "precision": {
+                    "type": "manual"
+                    }
+                }
+            },
+        "ignored_scopes": ['TwoConvTestModel/Sequential[features]/Sequential[0]/Conv2d[0]']
+        }
+    )
+    , model_creator=TwoConvTestModel, is_error=False),
+    FrozenLayersTestStruct(config=update_config(get_quantization_config_without_range_init(), {
+        "target_device": "VPU",
+        "compression": {
+            "algorithm": "quantization",
+            "initializer": {
+                "precision": {
+                    "type": "manual"
+                }
+            }
+        },
+        "target_scopes": ['TwoConvTestModel/Sequential[features]/Sequential[0]/Conv2d[0]']
+    })
+    , model_creator=TwoConvTestModel, is_error=True),
+    FrozenLayersTestStruct(config=get_quantization_config_without_range_init(),
+                           model_creator=TwoConvTestModel, is_error=False),
+    FrozenLayersTestStruct(config=update_config(get_quantization_config_without_range_init(), {
```

Comment:
I'd do it in another way.
```
class FrozenLayersTestStruct
    config_base=get_quantization_config_without_range_init
       
    def create_config():
```
for most of the cases, you just provide a config update, responsibility for creation of config is inside class.
```
FrozenLayersTestStruct(config_update={})
```
if you need another config_base (rare case), just override it.
```
FrozenLayersTestStruct(config_base=get_basic_pruning_config, config_update={})
```

Note, that you don't even need to create an instance of the final config, you just specify instructions on how to create it, final creation happens only once on `create_config` call

`config_base` is just a pointer to function, not a dictionary.